Sources/Conduit/Auth/TokenStorage/OAuth2TokenStore.swift

```diff
@@ -48,9 +77,46 @@ extension OAuth2TokenStore {
         }
     }
 
+    /// Removes a token from storage
+    ///
+    /// - Parameters:
+    ///   - client: Describes the token's OAuth2 client
+    ///   - authorization: The type and level of authorization that the token is used for
     public func removeTokenFor(client: OAuth2ClientConfiguration, authorization: OAuth2Authorization) {
         let emptyToken: BearerToken? = nil
         store(token: emptyToken, for: client, with: authorization)
     }
 
+    /// Determines if a refresh token is locked
+    ///
+    /// - Parameters:
+    ///   - client: Describes the token's OAuth2 client
+    ///   - authorization: The type and level of authorization that the token is used for
+    /// - Returns: A Bool indicating if the refresh token is locked
+    func isRefreshTokenLockedFor(client: OAuth2ClientConfiguration, authorization: OAuth2Authorization) -> Bool {
+        guard let expiration = refreshTokenLockExpirationFor(client: client, authorization: authorization) else {
+            return false
+        }
+        return Date() < expiration
+    }
+
+    func tokenIdentifierFor(clientConfiguration: OAuth2ClientConfiguration,
+                            authorization: OAuth2Authorization) -> String {
+        let authorizationLevel = authorization.level == .user ? "user-token" : "client-token"
+        return [
+            "com.mindbodyonline.connect.oauth-client",
```

Comment:
do we want this to read `com.mindbodyonline.connect` instead of a more generic key?

Reply:
Early mistake, but changing this would invalidate previous storage. It's meant to be `com.mindbodyonline.Conduit.oauth-client`

Reply:
Does there need to be a migration strategy for the previously stored tokens, or are just going to live with that file path name? (im okay with either, just curious)

Reply:
Usages of `OAuth2TokenDiskStore` with a `url` storage method will need to be migrated (retrieve the token and then re-store it to an `OAuth2TokenFileStore`). Usages of `OAuth2TokenDiskStore` with a `userDefaults` storage method can simply be replaced with `OAuth2TokenUserDefaultsStore(userDefaults: .standard)`

Reply:
But this is the token identifier generator for all the different Stores, which would mean all of the tokens would be stored with this piece always included in the identifier, right? Do we want something tied so closely to this hardcoded string? Especially since this may have wider distribution

Reply:
Good point -- I like the idea of making this extendable. For the purpose of this commit, I'd rather leave it for now (matches existing functionality), but I'll start looking at alternate approaches for this.